### ISSUE TITLE
Custom metric mix train and validation data

### ISSUE THREAD

**Issue by Kenvelo90** [tensorflow/tensorflow#25325]
**System information**

- Linux ubuntu, CPU
- installed packages using anaconda with python 3.6
- Tensorflow version v1.12.0-0-ga6d8ffae09 1.12.0
- Keras version  2.2.4

**Describe the current behavior**

When using a custom metric, the metric is computed on training data and validation data together.

**Describe the expected behavior**

The train metric should be on train data only and validation metric (*_val) should be on validation data only.

**Comment by ymodak** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
